understory_index/src/backends/rtree.rs

```diff
@@ -619,21 +623,40 @@ impl<T: Scalar, P: Copy + Debug> Backend<T> for RTree<T, P> {
         if let Some(old) = self.slots.get(slot).and_then(|x| *x)
             && let Some(root_idx) = self.root
         {
-            if Self::update_in_place(&mut self.arena, root_idx.get(), slot, old, aabb) {
+            if Self::update_in_place(
+                &mut self.arena,
+                &mut self.child_indices_arena,
+                root_idx.get(),
+                slot,
+                old,
+                aabb,
+            ) {
                 if let Some(s) = self.slots.get_mut(slot) {
                     *s = Some(aabb);
                 }
                 return;
             }
-            let _ = Self::search_remove(&mut self.arena, root_idx.get(), slot, &old);
+            let _ = Self::search_remove(
+                &mut self.arena,
+                &mut self.child_indices_arena,
+                root_idx.get(),
+                slot,
+                &old,
+            );
         }
         self.insert(slot, aabb);
     }
 
     fn remove(&mut self, slot: usize) {
         if let Some(old) = self.slots.get(slot).and_then(|x| *x) {
             if let Some(root_idx) = self.root {
-                let _ = Self::search_remove(&mut self.arena, root_idx.get(), slot, &old);
+                let _ = Self::search_remove(
+                    &mut self.arena,
+                    &mut self.child_indices_arena,
+                    root_idx.get(),
+                    slot,
+                    &old,
+                );
```

Comment:
Did you look at just creating a vec here and passing it down so that you get the per-call scratch buffer without having to store it on the tree? That gets you less benefit over multiple calls, but might get enough to be worthwhile? Not sure without measuring with a workload.